rockstor/rockstor-core
# During samba share restore, use share ID instead of share name. Fixes #2051 #2053

Fixes #2051 and fixes #2053. 

_**Note:**_ this PR is still a **draft** as it still requires testing. It is submitted as such to speed up discussion and edits of changes and their logic.

### Rationale
As described in the corresponding issue (#2051), restoring samba exports using the config Backup & Restore feature fails silently due to the fact that it still uses the share _name_ while the share **_ID_** is now required.

### Aims and logic
This PR aims at correcting this by using the share ID instead of its name for the samba export re-creation during the restore process.

### Tests
1. Create 3 samba exports as follows:
![image](https://user-images.githubusercontent.com/30297881/59160965-55332780-8aaa-11e9-9330-08d74442a42e.png)

2. Create a new config backup
![image](https://user-images.githubusercontent.com/30297881/59160974-7dbb2180-8aaa-11e9-87c2-0aa146a07fb9.png)

3. Delete samba exports from webUI
![image](https://user-images.githubusercontent.com/30297881/59160991-a216fe00-8aaa-11e9-96ec-8ddf22870d7f.png)


4. Go back to config backup and restore the previous config backup containing the three exports
![image](https://user-images.githubusercontent.com/30297881/59161023-19e52880-8aab-11e9-9114-3ff76fb26b86.png)

Note that the third export (`test_share02`) unexpectedly failed due to the following error:
```
[08/Jun/2019 23:20:32] DEBUG [storageadmin.views.config_backup:87] Trigger Samba share creation with e = {u'comment': u'Samba-Export', u'read_only': u'no', u'browsable': u'yes', u'snapshot_prefix': None, u'share': 10, 'shares': '10', u'shadow_copy': False, u'guest_ok': u'no', u'path': u'/mnt2/test_share02'}
[08/Jun/2019 23:20:33] DEBUG [storageadmin.views.samba:131] SambaListView request is = {u'comment': u'Samba-Export', u'read_only': u'no', u'browsable': u'yes', u'snapshot_prefix': None, u'share': 10, u'shares': u'10', u'shadow_copy': False, u'guest_ok': u'no', u'path': u'/mnt2/test_share02'}
[08/Jun/2019 23:20:33] ERROR [storageadmin.util:44] Exception: Operation not permitted on this share (home) because it is a special system share.
```

The shares IDs on this system are as follows:
![image](https://user-images.githubusercontent.com/30297881/59161071-ba3b4d00-8aab-11e9-9d32-e36e3ec64a4f.png)

As we can see above, although the correct ID (`10`) was sent to the api call, the export was attempted using the share id `1` (corresponding to the protected share `home`), indicating that parsing of share ID should be improved.


### Minor changes
Use .format() to print variables.
